### DISCUSSION TITLE
use wasm-pack during build

### DISCUSSION THREAD

**Discussion by majg0** [vercel/vercel#8235] [Help] [answered]
Hi! How do I use a custom tool like [wasm-pack](https://rustwasm.github.io/wasm-pack/) during my build?

More specifically, my setup requires my build to:

`wasm-pack build ./wasm --target web --release && mkdir static/.wasm && cp wasm/pkg/wasm* static/.wasm/`

and then `vite build` (which works out of the box, it is installed as an npm dep)

Am I looking to run `curl https://rustwasm.github.io/wasm-pack/installer/init.sh -sSf | sh` as part of a custom `build` script, as of https://rustwasm.github.io/wasm-pack/installer/ to enable this?

What's the correct approach here?

**Comment by amyegan** [comment, marked as answer]
Hi @majg0. I don't know much about Rust or WASM, but I'm trying to find other people who know more about it and might be able to help. 

In general, I recommend this guide to [using GitHub Actions with Vercel](https://vercel.com/support/articles/how-can-i-use-github-actions-with-vercel) and this [an example repo](https://github.com/vercel/examples/tree/main/ci-cd/github-actions) using GitHub Actions, which you can use for custom deployment automation.

**Reply by majg0** [reply]
Thanks, that does the trick nicely. I suppose I'd have to disable vercel's own git hook from running, and I assume I can do that from the dashboard. Will check this out later. You may close. Thanks for the support!

**Reply by amyegan** [reply]
Glad that helped! You can [disable Vercel for GitHub deployments](https://vercel.com/docs/project-configuration#git-configuration/github-enabled) in a `vercel.json` file in the root of your project.
```
{
  "github": {
    "enabled": false
  }
}
```
